### PR TITLE
Rotate geometry before calculating bounding box in atlas

### PR DESCRIPTION
**Pull request by Zverik**
When maps in the composer were rotated based on current atlas object, the bounding box for the object was calculated without taking the rotation into account. This should fix bug [11954](https://issues.qgis.org/issues/11954).

Tested locally, it worked.

### CODE REVIEW

**Comment by nyalldawson** [comment]
Nice fix for an annoying issue - thanks! Could you add a unit test for this in https://github.com/qgis/QGIS/blob/master/tests/src/python/test_qgsatlascomposition.py ? 

(by the way, #8466 and #8125 are totally different issues (not related to map rotation). They both relate to rotating any composer item, selecting multiple items, and then attempting to resize using the mouse drag handles)

**Review by nyalldawson — APPROVED** [review]

File: src/core/composer/qgsatlascomposition.cpp [hunk 1]
```diff
@@ -436,9 +436,13 @@ bool QgsAtlasComposition::prepareForFeature( const int featureI, const bool upda
...
 {
   // QgsGeometry::boundingBox is expressed in the geometry"s native CRS
-  // We have to transform the grometry to the destination CRS and ask for the bounding box
+  // We have to transform the geometry to the destination CRS and ask for the bounding box
   // Note: we cannot directly take the transformation of the bounding box, since transformations are not linear
-  mTransformedFeatureBounds = currentGeometry( map->crs() ).boundingBox();
+  QgsGeometry g( currentGeometry( map->crs() ) );
+  // Rotating the geometry, so the bounding box is correct wrt map rotation
+  if ( map->mapRotation() != 0.0 )
+    g.rotate( map->mapRotation(), g.centroid().asPoint() );
```
Comment (nyalldawson) [inline src/core/composer/qgsatlascomposition.cpp]: Is using the centroid of the geometry correct? I'm thinking possibly this should be the center of the geometry's original bounding box instead.

**Review by nyalldawson — CHANGES_REQUESTED** [review]
needs clarification regarding use of centroid

**Comment by Zverik** [comment]
I could not solve the issue of shifted map extend, so for now I grow the extent a little, so the feature is completely inside bounds. Experiments show that the margin is still very small, although for complex features it is noticeable. The math turned out too complex for me, sorry.

The python test is also done. Please take another look at this, @nyalldawson.

**Comment by 3nids** [comment]
I might not be the right guy for this one. 
@nyalldawson ok for you to merge? or someone else?

**Comment by nyalldawson** [comment]
Sorry for the delay - I'll take a look again tomorrow

**Comment by Zverik** [comment]
@nyalldawson ping :)

**Comment by nyalldawson** [comment]
Nice work - thank you!

**Pull request merged**